ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiUserManagementService.scala

```diff
@@ -210,4 +238,45 @@ object ApiUserManagementService {
       proto.Right(proto.Right.Kind.CanReadAs(proto.Right.CanReadAs(party)))
   }
 
+  def encodeNextPageToken(page: UsersPage): String =
+    page.lastUserIdOption
+      .map { id =>
+        val bytes = Base64.getUrlEncoder.encode(id.getBytes(StandardCharsets.UTF_8))
+        new String(bytes, StandardCharsets.UTF_8)
+      }
+      .getOrElse("")
+
+  def decodePageToken(pageToken: String)(implicit
+      loggingContext: ContextualizedErrorLogger
+  ): Either[StatusRuntimeException, Option[Ref.UserId]] = {
+    if (pageToken.isEmpty) {
+      Right(None)
+    } else {
+      val bytes = pageToken.getBytes(StandardCharsets.UTF_8)
+      for {
+        decodedBytes <- Try[Array[Byte]](Base64.getUrlDecoder.decode(bytes))
+          .map(Right(_))
+          .recover { case _: IllegalArgumentException =>
+            Left(
+              LedgerApiErrors.RequestValidation.InvalidArgument
+                .Reject("Invalid page token")
+                .asGrpcError
+            )
+          }
+          .get
+        decodedStr = new String(decodedBytes, StandardCharsets.UTF_8)
```

Comment:
@rautenrieth-da : this is a good point. I believe the motivation for using protobuf is to allow evolving the format of page tokens across server versions without breaking the state of clients that receive a token from the old server and then submit a request to the new server.

Moreover, it ensure that the robust deserialization code from protobuf is used by instead of homebrewn one.

I'd be inclined to spend the effort of establishing the pattern of using protobuf for page-tokens here. I'd expect that to be an extra proto file living alongside the ApiServer classes, which contains a message for each token. Here, `ListUsersPageToken`.

@rautenrieth-da @mziolekda @pbatko-da : do you think it is worth the effort to establish the pattern now, and use it consistently going forward when sending back page tokens?

Reply:
I'm for trying it out now, albeit in a separate PR. 
We should get a more complete understanding of what it means to support pagination (especially in view of pagination vs. streaming API discussion).

Reply:
@pbatko-da Please create a JIRA ticket and then close this thread

Reply:
https://digitalasset.atlassian.net/browse/DPP-888

Reply:
Draft PR: https://github.com/digital-asset/daml/pull/12764